packages/@aws-cdk/cicd/README.md

```diff
@@ -0,0 +1,112 @@
+
+### Getting Started
+In order to add the `PipelineDeployStackAction` to your *CodePipeline*, you need to have a *CodePipeline* artifact that
+contains the result of invoking `cdk synth -o <dir>` on your *CDK App*. You can for example achieve this using a
+*CodeBuild* project.
+
+The example below defines a *CDK App* that contains 3 stacks:
+* `CodePipelineStack` manages the *CodePipeline* resources, and self-updates before deploying any other stack
+* `ServiceStackA` and `ServiceStackB` are service infrastructure stacks, and need to be deployed in this order
+
+```
+  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━┓  ┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
+  ┃     Source     ┃  ┃     Build      ┃  ┃  Self-Update   ┃  ┃            Deploy             ┃
+  ┃                ┃  ┃                ┃  ┃                ┃  ┃                               ┃
+  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┃  ┃ ┌────────────┐ ┌────────────┐ ┃
+  ┃ │   GitHub   ┣━╋━━╋━▶ CodeBuild  ┣━╋━━╋━▶Deploy Stack┣━╋━━╋━▶Deploy Stack┣━▶Deploy Stack│ ┃
+  ┃ └────────────┘ ┃  ┃ └────────────┘ ┃  ┃ └────────────┘ ┃  ┃ └────────────┘ └────────────┘ ┃
+  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━┛  ┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
+```
+
+#### `index.ts`
+```ts
+import codebuild = require('@aws-cdk/aws-codebuild');
+import codepipeline = require('@aws-cdk/aws-codepipeline');
+import cdk = require('@aws-cdk/cdk');
+import cicd = require('@aws-cdk/cicd');
+
+const app = new cdk.App();
+
+// We define a stack that contains the CodePipeline
+const pipelineStack = new cdk.Stack(app, 'CodePipelineStack');
+const pipeline = new codepipeline.Pipeline(pipelineStack, 'CodePipeline', {
+  restartExecutionOnUpdate: true,
+  /* ... */
+});
+// Configure the CodePipeline source - where your CDK App's source code is hosted
+const source = new codepipeline.GitHubSourceAction(pipelineStack, 'GitHub', {
+  stage: pipeline.addStage('source'),
+  /* ... */
+});
+const project = new codebuild.Project(pipelineStack, 'CodeBuild', {
+  artifacts: new codebuild.CodePipelineArtifacts(),
+  source: new codebuild.CodePipelineSource(),
+  /* ... */
+});
+const synthesizedApp = project.outputArtifact;
+
+// Optionally, self-update the pipeline stack
+const selfUpdateStage = pipeline.addStage('SelfUpdate');
+new cicd.PipelineDeployStackAction(pipelineStack, 'SelfUpdatePipeline', {
+  stage: selfUpdateStage,
+  stack: pipelineStack,
+  inputArtifact: synthesizedApp,
+});
+
+// Now add our service stacks
+const deployStage = pipeline.addStage('Deploy');
+const serviceStackA = new MyServiceStackA(app, 'ServiceStackA', { /* ... */ });
+const serviceStackB = new MyServiceStackB(app, 'ServiceStackB', { /* ... */ });
+// Add actions to deploy the stacks in the deploy stage:
+new cicd.PipelineDeployStackAction(pipelineStack, 'DeployServiceStack', {
+  stage: deployStage,
+  stack: serviceStackA,
+  inputArtifact: synthesizedApp,
+  createChangeSetRunOrder: 1,
```

Comment:
Is this really needed? If not, omit. It's useful the README examples to be as minimal as possible to help people get started.

Reply:
I also try to avoid them being "too minimal" (aka it's a trivial example but absolutely doesn't help me achieve my non-trivial use-case). I remove the default-ed stuff though.